MediaBrowser.Common/Net/NetworkExtensions.cs

```diff
@@ -56,7 +56,23 @@ public static bool IsIPv6LinkLocal(IPAddress address)
         /// <returns>String value of the subnet mask in dotted decimal notation.</returns>
         public static IPAddress CidrToMask(byte cidr, AddressFamily family)
         {
-            uint addr = 0xFFFFFFFF << (family == AddressFamily.InterNetwork ? 32 : 128 - cidr);
+            uint addr = 0xFFFFFFFF << ((family == AddressFamily.InterNetwork ? 32 : 128) - cidr);
+            addr = ((addr & 0xff000000) >> 24)
```

Comment:
Code duplication

Reply:
Not sure how to make this better though. Would you prefer to hav th whole line 59 in every of the next lines?

Reply:
It's fine